public/components/main/main.tsx

```diff
@@ -64,11 +79,182 @@ export class Main extends React.Component<RouterHomeProps, any> {
+
+    return (
+      <div>
+        <EuiOverlayMask>
+          <EuiModal onClose={closeModal}>
+            <EuiModalHeader>
+              <EuiTitle>
+                <EuiText textAlign="right">
+                  <h2>Generating report</h2>
+                </EuiText>
+              </EuiTitle>
+            </EuiModalHeader>
+            <EuiModalBody>
+              <EuiText>Preparing your file for download...</EuiText>
+              <EuiSpacer />
+              <EuiFlexGroup justifyContent="spaceAround" alignItems="center">
+                <EuiFlexItem>
+                  <EuiLoadingSpinner size="xl" />
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </EuiModalBody>
+          </EuiModal>
+        </EuiOverlayMask>
+      </div>
+    );
+  }
+
+  readStreamToFile = async (stream: string) => {
+    var link = document.createElement('a');
+    var fileFormatPrefix = getFileFormatPrefix(
+      this.state.generateReportFileFormat
+    );
+    var url = fileFormatPrefix + stream;
+    if (typeof link.download == 'string') {
+      if (fileFormatPrefix.includes('pdf')) {
+        link.download = this.state.generateReportFilename.substring(
+          0,
+          this.state.generateReportFilename.length - 4
+        );
+      } else {
+        link.download = this.state.generateReportFilename;
+      }
+      link.href = url;
+      document.body.appendChild(link);
+      link.click();
+      await this.updateReportsTableContent();
+      document.body.removeChild(link);
+    } else {
+      window.open(url, '_blank');
+    }
+  };
+
+  generateReport = async (metadata) => {
+    this.setState({
+      showGenerateReportLoadingToast: true,
+    });
+    const retVal = fetch('../api/reporting/generateReport', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'kbn-xsrf': 'reporting',
+      },
+      body: JSON.stringify(metadata),
+    })
+      .then(async (resp) => {
+        this.setState({
+          generateReportFilename: extractFilename(resp.headers),
+        });
+        this.setState({
+          generateReportFileFormat: extractFileFormat(resp.headers),
+        });
+        this.readStreamToFile(await resp.text());
+        this.setState({
+          showGenerateReportLoadingToast: false,
+        });
+        return resp;
+      })
+      .catch((error) => {
+        console.log('printing error: ', error);
+      });
+  };
+
+  updateReportsTableContent = async () => {
+    const { httpClient } = this.props;
+    httpClient
+      .get('../api/reporting/reports')
+      .then((response) => {
+        this.setState({
+          reportsTableContent: addReportsTableContent(response.data),
+        });
+      })
+      .catch((error) => {
+        console.log('error when fetching all reports: ', error);
+      });
+  };
+
   componentDidMount() {
     const { httpClient } = this.props;
-    httpClientGlobal = httpClient;
+    // get all reports
+    httpClient
```

Comment:
let's covert to await/async

  componentDidMount = async () => {
    const { httpClient } = this.props;
    try {
       const reportResponse = await httpClient.get('../api/reporting/reports')
       const reportDefinitionReponse = await httpClient.get('../..');
    } catch (e) {
    }
  };

Reply:
Will change